### ISSUE TITLE
Maintainence status (tests failing for the past 6 months)

### ISSUE THREAD

**Issue by adrinjalali** [adriangb/scikeras#329]
Looking at the jobs run everyday, it seems the last time tests passed was about 6 months ago, after which, there hasn't been any activity on the repo. 

I was wondering if the repo is still active and y'all intend to maintain it. I'm also not sure if `keras` intends to put back these wrappers now that it's a standalone package.

**Comment by adriangb** [comment]
The issues are that I don't really use TensorFlow anymore and the TensorFlow team has historically made breaking changes in minor releases.

I've been giving this some thought myself and would really love for someone else who is actively using this software to take over maintenance of it.

**Comment by adrinjalali** [comment]
Those are very fair points. I'll wait and see what the Keras team says, and I'll raise this internally with our team to see if we'd have capacity to help.

**Comment by clstaudt** [comment]
I have found the scikeras wrappers very useful in my ML work, which regularly combines deep learning and "shallow learning" ("classical ML") models. I was surprised that the tensorflow project has removed them, happy to find them again in scikeras, and I'd be happy to help with keeping this project maintained.

**Comment by adriangb** [comment]
@clstaudt amazing thank you! maybe a good place to start would be to get nightly CI back to passing?

**Comment by clstaudt** [comment]
@adriangb  Here's a start - please review
https://github.com/adriangb/scikeras/pull/330

**Comment by adrinjalali** [comment]
Note that in https://github.com/keras-team/keras/issues/20399 folks agreed to have these wrappers in keras itself, and I was working on cleaning up the code from here to port it there. Would we rather do that and maintain the code there (which means we'd have to support a single version of `keras` then), or fix the issues here?

**Comment by clstaudt** [comment]
@adrinjalali Why would they first remove them and then want them back in keras? Am I missing something or is this just regular chaos?

I think it makes sense that the keras project maintains a scikit-learn compatibility layer.

**Comment by adrinjalali** [comment]
Keras used to be an independent project, providing a nice API layer on top of TF and Caffe (if I'm not mistaken). Then it was adopted by and moved inside TF by google, and became a TF thingy. 

Now TF is moving things out of the TF repo I suppose since it's become a VERY large project and it makes sense to have different modules in their own repos. Keras is also supposed to now support TF, PyTorch, and Jax.

So I can imagine the vibe in keras is now very different and understand that decisions are made differently. I'm personally happy about this development.

**Comment by clstaudt** [comment]
@adrinjalali  Since you have agreed to work on the migration back to keras: Thanks, and feel free to delegate some of the easier tasks if you need help.

**Comment by adriangb** [comment]
I think it would be best if this moved back into Keras, as long as they're willing to accept it. Thank you @adrinjalali for getting the ball rolling on that!

**Comment by adriangb** [comment]
@clstaudt I do think your PR is still worth pursuing. It will be easier to port this to Keras if we know things are working and passing with the latest version of Keras, which will be easier to do here if nothing else because CI is faster, etc.

**Comment by clstaudt** [comment]
@adriangb  I agree, but after trying to fix some of the failing tests I realize that I am not able to complete this independently. I could suggest incremental fixes and work with you through review, but that requires your time as well. Perhaps more than you would need to fix the couple of errors yourself, knowing the code base.

**Comment by adriangb** [comment]
okay no problem 😄, I'll see if I have time to pick it up in the next couple weeks otherwise we'll see how the conversation goes on the Keras side.

**Comment by adrinjalali** [comment]
PR on the keras side: https://github.com/keras-team/keras/pull/20599

**Comment by adrinjalali** [comment]
Since https://github.com/keras-team/keras/pull/20599 is now merged, we might want to slowly refer people to there in this repo?

**Comment by adriangb** [comment]
Yes agreed! Do you have any suggestions on how to do that? I propose:

- Put a warning on import that explains that this package will no longer be maintained
- Put a warning in the README and docs that explains that this package will no longer be maintained
- Mark the repository as archived

**Comment by adrinjalali** [comment]
That sounds like a perfect plan to me. I'm only not sure if we should wait till the next Keras release to have the wrappers available to users?

**Comment by adriangb** [comment]
Sure no harm in waiting 😄 at least to release a version with the warning. I've set a reminder for 2 weeks from now (taking a guess at it based on release cadence) but feel free to remind me. Nothing stopping us from preparing the PR to merge here though.

**Comment by adriangb** [comment]
https://github.com/adriangb/scikeras/pull/334

**Comment by clstaudt** [comment]
Thanks @adrinjalali  - any idea about when this will be in a keras relase?

**Comment by adrinjalali** [comment]
looking at the cadence, probably early Jan. But I'm not sure since I'm not a keras maintainer.
